src/fides/api/ops/schemas/user_permission.py

```diff
@@ -31,10 +31,29 @@ def validate_scopes(cls, scopes: List[str]) -> List[str]:
             )
         return scopes
 
+    @validator("roles")
+    @classmethod
+    def validate_roles(cls, roles: List[str]) -> List[str]:
+        """Validates that all incoming roles are valid"""
+        diff: Set = set(roles).difference(set(ROLE_REGISTRY))
+        if len(diff) > 0:
+            raise HTTPException(
+                status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=f"Invalid Role(s) {diff}. Roles must be one of {ROLE_REGISTRY}.",
+            )
+        return roles
+
+
+class UserPermissionsEdit(UserPermissionsCreate):
+    """Data required to edit a FidesUserPermissions record."""
+
+    id: Optional[
+        str
+    ]  # I don't think this should be in the request body, so making it optional.
```

Comment:
The UI currently sends an id in with the edit user permission's request.  I don't think this needs to be there, so I'm making it optional here and am ensuring we ignore it.  There was no actual validation on that value, so if we weren't ignoring it, it could technically replace the id of the user permissions object.

It's fine as it's not being used, but at some point, it would be nice to remove this id from the request body in the UI too.